packages/react-components/src/components/AppFrame/components/SideNavigationGroup/SideNavigationGroup.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+
+import { ChevronRight } from '@livechat/design-system-icons';
+import cx from 'clsx';
+
+import noop from '../../../../utils/noop';
+import { Icon } from '../../../Icon';
+import { Text } from '../../../Typography';
+import { SideNavigationItem } from '../SideNavigationItem/SideNavigationItem';
+
+import { ISideNavigationGroupProps } from './types';
+
+import styles from './SideNavigationGroup.module.scss';
+
+const baseClass = 'side-navigation-group';
+
+export const SideNavigationGroup: React.FC<ISideNavigationGroupProps> = ({
+  label,
+  rightNode,
+  className,
+  children,
+  isCollapsible,
+  onItemHover,
+  shouldOpenOnInit = false,
+  shouldOpenOnActive = false,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(
+    !isCollapsible || shouldOpenOnInit
+  );
+  const [isSideMenuVisible, setIsSideMenuVisible] = React.useState(false);
+  const [hasActiveElements, setHasActiveElements] =
+    React.useState<boolean>(false);
+  const hadActiveListElementsRef = React.useRef(false);
+  const localRightNode =
+    typeof rightNode === 'function' ? rightNode(isOpen) : rightNode;
+  const localLabel = typeof label === 'function' ? label(isOpen) : label;
+
+  const openList = (): void => setIsOpen(true);
+
+  const toggle = (): void => {
+    if (!isCollapsible) return;
+    setIsOpen((prev) => !prev);
+  };
+
+  const onSetListNode = (node: HTMLUListElement): void => {
+    const listElements = node?.getElementsByTagName('li');
+    const hasListActiveElements = listElements?.length
+      ? [...listElements].some(
+          (el) => el.getAttribute('data-active') === 'true'
+        )
+      : false;
+
+    setHasActiveElements(hasListActiveElements);
+  };
+
+  React.useEffect(() => {
+    if (!shouldOpenOnActive) {
+      return;
+    }
+
+    if (!hadActiveListElementsRef.current && hasActiveElements) {
+      openList();
+    }
+
+    hadActiveListElementsRef.current = hasActiveElements;
+  }, [hasActiveElements, shouldOpenOnActive]);
+
+  React.useEffect(() => {
+    if (!isCollapsible) {
+      return setIsSideMenuVisible(true);
+    }
+
+    if (isOpen) {
+      return setIsSideMenuVisible(true);
+    }
+
+    return setIsSideMenuVisible(false);
+  }, [isOpen]);
```

Comment:
Why `isCollapsible` is not in deps array?

The hook can be optimized to just this:

```suggestion
  React.useEffect(() => {
    return setIsSideMenuVisible(!isCollapsible || isOpen);
  }, [isOpen]);
```